govcd/vapp.go

```diff
@@ -100,6 +100,14 @@ func (vapp *VApp) Refresh() error {
 // acceptAllEulas - setting allows to automatically accept or not Eulas.
 func (vapp *VApp) AddVM(orgVdcNetworks []*types.OrgVDCNetwork, vappNetworkName string, vappTemplate VAppTemplate, name string, acceptAllEulas bool) (Task, error) {
 
+	if vappTemplate == (VAppTemplate{}) || vappTemplate.VAppTemplate == nil {
+		return Task{}, fmt.Errorf("vApp Template can not be empty")
+	}
+
+	if vappTemplate.VAppTemplate.Status != 8 { // status Ready
```

Comment:
Would be really good to confirm all available status codes and put them into comment.

Reply:
Comments added.